src/App-routes.spec.ts

```diff
@@ -1012,6 +1028,84 @@ describe('App event routing', () => {
       // Assert
       assertMiddlewaresNotCalled();
     });
+
+    it('should handle bot_message events', async () => {
+      // Act
+      app.message(PASS_STRING, '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'bot_message', message);
+      // Assert
+      assertMiddlewaresCalledOnce();
+      assertMiddlewaresCalledOrder();
+    });
+    it('should not handle bot_message events when the constraints do not match', async () => {
+      // Act
+      app.message('foo-bar', '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'bot_message', message);
+      // Assert
+      assert.isFalse(fakeMiddleware1.calledOnce);
+    });
+    it('should handle file_share events', async () => {
+      // Act
+      app.message(PASS_STRING, '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'file_share', message);
+      // Assert
+      assertMiddlewaresCalledOnce();
+      assertMiddlewaresCalledOrder();
+    });
+    it('should not handle file_share events when the constraints do not match', async () => {
+      // Act
+      app.message('foo-bar', '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'file_share', message);
+      // Assert
+      assert.isFalse(fakeMiddleware1.calledOnce);
+    });
+    it('should handle thread_broadcast events', async () => {
+      // Act
+      app.message(PASS_STRING, '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'thread_broadcast', message);
+      // Assert
+      assertMiddlewaresCalledOnce();
+      assertMiddlewaresCalledOrder();
+    });
+    it('should not handle message_changed events', async () => {
+      // Act
+      app.message(PASS_STRING, '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'message_changed', message);
+      // Assert
+      assert.isFalse(fakeMiddleware1.calledOnce);
+    });
+    it('should handle message_changed events when using allMessageSubtypes', async () => {
+      // Act
+      app.allMessageSubtypes(PASS_STRING, '- val', ...fakeMiddlewares);
+      await fakeSubtypedMessageEvent(fakeReceiver, 'message_changed', message);
+      // Assert
+      assertMiddlewaresCalledOnce();
+      assertMiddlewaresCalledOrder();
+    });
+
+    it('should provide better typed payloads', async () => {
+      app.message(async ({ payload }) => {
+        // verify it compiles
+        assert.isNotNull(payload.channel);
```

Comment:
Just an idea: many of the `assert` methods take a type parameter that then ensures the type of the argument provided to `assert` is of the provided type. This way perhaps it is clearer the intention of the test ("verify it compiles").

```suggestion
        assert.isNotNull<string>(payload.channel);
```